### PR TITLE
Add samples for cloud storage json api.

### PR DESCRIPTION
**Pull request by jerjou**
Move in the sample from [its own repo](https://github.com/GoogleCloudPlatform/cloud-storage-docs-json-api-examples/tree/master/java-example), modify to use Application Default Credentials, and add samples for uploading and deleting objects

### CODE REVIEW

File: cloud-storage/json-api/src/main/java/StorageSample.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
+//[START all]
+/*
+ * Copyright (c) 2014 Google Inc.
```
Comment (waprin) [inline cloud-storage/json-api/src/main/java/StorageSample.java]: The ultimate nit...2015

File: cloud-storage/json-api/src/main/java/StorageSample.java [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+import java.io.InputStream;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Main class for the Cloud Storage API command line sample.
+ * Demonstrates how to make an authenticated API call using Application Default
+ * Credentials.
```
Comment (waprin) [inline cloud-storage/json-api/src/main/java/StorageSample.java]: formatting

File: cloud-storage/json-api/src/main/java/StorageSample.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+  /**
+   * Fetches the metadata for the given bucket.
+   *
+   * @param bucketName the name of the bucket to get metadata about.
+   * @return a Bucket containing the bucket's metadata.
+   */
+  public static Bucket getBucket(String bucketName) throws IOException, GeneralSecurityException {
+    Storage client = getService();
+
+    Storage.Buckets.Get bucketRequest = client.buckets().get(bucketName);
```
Comment (waprin) [inline cloud-storage/json-api/src/main/java/StorageSample.java]: maybe add a small comment here or in the javadoc what setProjection does?

File: cloud-storage/json-api/src/main/java/StorageSample.java [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+  // [START list_bucket]
+  /**
+   * Fetch a list of the objects within the given bucket.
+   *
+   * @param bucketName the name of the bucket to list.
+   * @return a list of the contents of the specified bucket.
+   */
+  public static List<StorageObject> listBucket(String bucketName)
+      throws IOException, GeneralSecurityException {
```
Comment (waprin) [inline cloud-storage/json-api/src/main/java/StorageSample.java]: guava checkNotNull?

File: cloud-storage/json-api/pom.xml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+          <mainClass>StorageSample</mainClass>
+        </configuration>
+      </plugin>
+    </plugins>
+    <finalName>${project.artifactId}-${project.version}</finalName>
+  </build>
+  <dependencies>
+    <dependency>
+      <groupId>com.google.apis</groupId>
+      <artifactId>google-api-services-storage</artifactId>
```
Comment (waprin) [inline cloud-storage/json-api/pom.xml]: why not gcloud-java?

**Comment by waprin** [comment]
Mostly LGTM to me, nice test coverage, added some nits/opinions.

File: cloud-storage/json-api/pom.xml (hunk 5 above)
Comment (jerjou) [inline cloud-storage/json-api/pom.xml]: Hm... I think maybe we should do a larger pass over this whole repo to migrate to gcloud-java, after it stabilizes a bit.

**Pull request merged**